src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -0,0 +1,954 @@
+            Metadata.CustomMetadata customMd = new Metadata.CustomMetadata();
+
+            if (this.targetObjectMappingLabel != null && this.dataImportObjectName != null) {
+                customMd.label = this.label.replace(
+                    this.dataImportObjectName,
+                    this.targetObjectMappingLabel).left(40);
+            }
+
+            if (this.recordName == null) {
+                this.recordName = generateRecordName(
+                    this.label,
+                    true,
+                    this.targetObjectMappingLabel,
+                    this.dataImportObjectName);
+            }
+
+            customMd.fullName = mdTypeName + '.' + this.recordName.left(40).removeEnd('_');
+            customMd.protected_x = false;
+
+            // Convert an empty string ('') value into null for the entity reference to save properly.
+            if (String.isEmpty(this.dataImportFieldMappingSetRecordName)) {
+                this.dataImportFieldMappingSetRecordName = null;
+            }
+
+            // Convert an empty string ('') value into null for the entity reference to save properly.
+            if (String.isEmpty(this.targetObjectMappingRecordName)) {
+                this.targetObjectMappingRecordName = null;
+            }
+
+            Map<String, Object> fieldValues;
+            if (this.isDeleted != true) {
+
+                if (NAMESPACE != 'npsp') {
+                    this.targetFieldAPIName =
+                        UTIL_Namespace.removeNSPrefixNpspOnly(this.targetFieldAPIName);
+                }
+
+                fieldValues = new Map<String, Object>{
+                    'Data_Import_Field_Mapping_Set__c' => dataImportFieldMappingSetRecordName == null ? 'Default_Field_Mapping_Set' : dataImportFieldMappingSetRecordName,
+                    'Required__c' => isRequired,
+                    'Source_Field_API_Name__c' => sourceFieldAPIName,
+                    'Target_Field_API_Name__c' => targetFieldAPIName,
+                    'Target_Object_Mapping__c' => targetObjectMappingRecordName,
+                    'Is_Deleted__c' => isDeleted
+                };
+            } else {
+                fieldValues = new Map<String, Object>{ 'Is_Deleted__c' => isDeleted };
+            }
+            customMd = CMT_MetadataAPI.addFieldsToMetadataObject(customMd, fieldValues);
+
+            return customMd;
+        }
+
+        /****************************************************************************************************
+        * @description Utility to convert a name into a valid 'DeveloperName' quality value (or unique GUID)
+        * for the CMT record
+        *
+        * @param name String
+        * @param makeUnique Boolean
+        * @return DeveloperName format string
+        */
+        private String generateRecordName(String name, Boolean makeUnique, String objectLabel, String objectAPIName) {
```

Comment:
Can we avoid duplicating this between BDI_ManageAdvancedMappingCtrl.cls and here?